### ISSUE TITLE
Dependencies failing

### ISSUE THREAD

**Issue by AhtiAhde** [rsomani95/shot-type-classifier#3]
After Pytorch 1.14.0 came out, something changed.

After runninng:
```
pip install http://download.pytorch.org/whl/cpu/torch-1.0.0-cp36-cp36m-linux_x86_64.whl
pip install fastai
```
I get error `ERROR: torchvision 0.5.0 has requirement torch==1.4.0, but you'll have torch 1.0.0 which is incompatible.`, after which running the predictions will result in: `ImportError: cannot import name 'Optional'`.

After checking all available versions of fastai, one can see that new version of pip only allows you to install 1.0.60.

Fastai 1.0.60 works with torchvision 0.5.0, which requires torch 1.4.0, but that leads to many bugs.

We would need to make a pull request for fixing that.

EDIT: My bad, the newest pip does provide other version of fastai; yolk only tells the newest version for some reason. I will try to figure out working configuration and let you know if I do.

**Comment by rsomani95** [comment]
That's odd. Try doing a developer install of fastai:
```bash
git clone https://github.com/fastai/fastai
cd fastai
tools/run-after-git-clone
pip install -e ".[dev]"
```

I'm running `torch=1.3.1`, `torchvision=0.5.0a0` (source install for some odd reason), and `fastai=1.0.61dev0`

**Comment by AhtiAhde** [comment]
Hmm... in the installation instructions `pip install http://download.pytorch.org/whl/cpu/torch-1.0.0-cp36-cp36m-linux_x86_64.whl` will give you pytorch 1.0.0 and this used to work with that. For some reason I get different errors if pytorch is something else.

I am running on a machine with CUDA and trying to use the CPU version of this lib as it seems to cause memory leaks with CUDA enabled.

**Comment by AhtiAhde** [comment]
Which pip version are you running with?

**Comment by AhtiAhde** [comment]
To me pip demands torch 1.4.0 for torchvision 0.5.0

**Comment by rsomani95** [comment]
How strange. I'm using `pip=19.3.1`. 
Let me try it out on a CPU system and get back to you. Meanwhile, can you try doing a developer install of `fastai`. The installation instructions I've linked to seem outdated.

**Comment by AhtiAhde** [comment]
I also have `pip=19.3.1`.

It seems like fastai wants `torchvision==0.5.0`, which needs `torch==1.4.0`. In order to downgrade for `torch==1.3.1` I had to also `pillow==6.1` and now it seems to work at least on my dev machine.

With the production machine, that has GPU, everything worked fine after we changed:
`pip install http://download.pytorch.org/whl/cpu/torch-1.0.0-cp36-cp36m-linux_x86_64.whl`
`pip install fastai`
to simply:
`pip install fastai torch`

Thanks for your time, perhaps there is no need for further fixes.
